### ISSUE TITLE
Version 1.5.0 breaks argpare.parse_args(namespace)

### ISSUE THREAD

**Issue by kovidgoyal** [python/mypy#15847]
Here is a test case:
```py
import argparse

class Options(argparse.Namespace):
    something: bool = True

p = argparse.ArgumentParser()
a: Options = p.parse_args(namespace=Options())
```

Running mypy version 1.5.0 on it gives the spurious error
```
10: error: Incompatible types in assignment (expression has type "Namespace", variable has type "Options")  [assignment]
    a: Options = p.parse_args(namespace=Options())
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```

Running mypy version 1.3.0 on it passes.

**Comment by cdce8p** [comment]
I bisected the issue to 59a6c5e73. Seems like an issue with the `argparse` typing stubs.
Looks like it's already fixed in `typeshed`, however the sync didn't make it in time for this release. Probably in `1.6.0` then.
- https://github.com/python/typeshed/pull/10387
- https://github.com/python/mypy/pull/15792

CC @AlexWaygood

**Comment by AlexWaygood** [comment]
Yeah, the issue was cause by https://github.com/python/typeshed/pull/10307, which made it into mypy v1.5. It's already been fixed on typeshed's `main` branch by https://github.com/python/typeshed/pull/10387, but the fix unfortunately _didn't_ make it into mypy v1.5.

Since https://github.com/python/typeshed/pull/10387 was a fairly small PR, it could possibly be cherry-picked into the next mypy patch release. However, I know @svalentin has already created the mypy 1.5.1 tag, so I'm not sure if it's possible to get a fix into mypy 1.5.1 :/

(We also can't create any new mypy releases at all until https://github.com/pypi/support/issues/3094 is resolved)

**Comment by alicederyn** [comment]
This appears to be fixed in mypy 1.6.0.
